reorderable/src/commonMain/kotlin/org/burnoutcrew/reorderable/Reorderable.kt

```diff
@@ -15,69 +15,38 @@
  */
 package org.burnoutcrew.reorderable
 
+import androidx.compose.foundation.gestures.awaitEachGesture
+import androidx.compose.foundation.gestures.awaitFirstDown
 import androidx.compose.foundation.gestures.drag
-import androidx.compose.foundation.gestures.forEachGesture
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.geometry.Offset
-import androidx.compose.ui.input.pointer.PointerId
-import androidx.compose.ui.input.pointer.PointerInputChange
-import androidx.compose.ui.input.pointer.PointerInputScope
 import androidx.compose.ui.input.pointer.changedToUp
 import androidx.compose.ui.input.pointer.pointerInput
 import androidx.compose.ui.input.pointer.positionChange
-import androidx.compose.ui.util.fastFirstOrNull
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.layout.positionInWindow
 
 fun Modifier.reorderable(
     state: ReorderableState<*>
 ) = then(
-    Modifier.pointerInput(Unit) {
-        forEachGesture {
-            val dragStart = state.interactions.receive()
-            val down = awaitPointerEventScope {
-                currentEvent.changes.fastFirstOrNull { it.id == dragStart.id }
-            }
-            if (down != null && state.onDragStart(down.position.x.toInt(), down.position.y.toInt())) {
-                dragStart.offset?.apply {
-                    state.onDrag(x.toInt(), y.toInt())
+    Modifier.onGloballyPositioned { state.layoutWindowPosition.value = it.positionInWindow()}.pointerInput(Unit) {
```

Comment:
You can simplify this solution using `androidx.compose.foundation.gestures` `detectDragGestures` and `detectDragGesturesAfterLongPress`

```kotlin
fun Modifier.reorderable(
    state: ReorderableState<*>
): Modifier = this then Modifier.onGloballyPositioned {
    state.layoutWindowPosition = it.positionInWindow()
}
```